### PR TITLE
Tidy delegate registration

### PR DESCRIPTION
**Pull request by duckranger**

### CODE REVIEW

File: src/Reshuffle.ts [hunk 1]
```diff
@@ -63,10 +63,8 @@ export default class Reshuffle {
     let httpMultiplexer = this.httpDelegates[path]
```
Comment (lagache) [inline src/Reshuffle.ts:63]:
```suggestion
    const httpMultiplexer = this.httpDelegates[path] || new HttpMultiplexer(path)
    httpMultiplexer.delegates.push(delegate)
```

**Review by ashevat — APPROVED** [review]

File: src/Reshuffle.ts [hunk 2]
```diff
@@ -60,13 +60,12 @@ export default class Reshuffle {
   }
 
   registerHTTPDelegate(path: string, delegate: BaseHttpConnector): Reshuffle {
-    let httpMultiplexer = this.httpDelegates[path]
+    const httpMultiplexer = this.httpDelegates[path] || new HttpMultiplexer(path)
```
Comment (ashevat) [inline src/Reshuffle.ts:63]: If this is const then the assignment in a few lines fails.

**Review by ashevat — CHANGES_REQUESTED** [review]

File: src/Reshuffle.ts [hunk 3]
```diff
@@ -60,13 +60,12 @@ export default class Reshuffle {
...
+    httpMultiplexer.delegates.push(delegate)
     if (!httpMultiplexer) {
```
Comment (lagache) [inline src/Reshuffle.ts:65]: This if needs to be deleted as it is done line 63

File: src/Reshuffle.ts (hunk 2 above)
Comment (lagache) [reply src/Reshuffle.ts:63]: line 65 and 67 needs to be deleted, catered by the `|| new HttpMultiplexer(path)`, and so the const will remain ;)

Comment (ashevat) [reply src/Reshuffle.ts:63]: @lagache  see build error:
src/Reshuffle.ts:66:7 - error TS2588: Cannot assign to 'httpMultiplexer' because it is a constant.

Comment (lagache) [reply src/Reshuffle.ts:63]: @ashevat yes I know, sorry I made one mistake in my comment above, line 65 **TO** 67 need to be removed.
the if is replaced by the `|| new HttpMultiplexer(path)`
@duckranger do you confirm?

**Review by lagache — APPROVED** [review]

**Pull request merged**